backend/coreapp/views/scratch.py

```diff
@@ -22,13 +22,18 @@
 import zipfile
 import re
 
-from ..models import Asm, Scratch
+from ..models import Asm, Project, ProjectFunction, Scratch
 from ..middleware import Request
 from ..decorators.django import condition
 from ..serializers import TerseScratchSerializer, ScratchCreateSerializer, ScratchSerializer
+from ..github import GitHubRepo, GitHubRepoBusy
 
 logger = logging.getLogger(__name__)
 
+class ProjectNotMaintainer(APIException):
```

Comment:
all exception classes should end in Exception imo

Reply:
also I'd call this NotProjectMaintainerException cause "Project not maintainer" sounds like "the project is not the maintainer" to me